pkgs/tools/misc/calamares/nixos-extensions-paths.patch

```diff
@@ -0,0 +1,46 @@
+diff --git a/src/calamares/main.cpp b/src/calamares/main.cpp
+index de709156f..a0b6c5faf 100644
+--- a/src/calamares/main.cpp
++++ b/src/calamares/main.cpp
+@@ -131,6 +132,8 @@ main( int argc, char* argv[] )
+     // TODO: umount anything in /tmp/calamares-... as an emergency save function
+ #endif
+ 
++    CalamaresUtils::setNixosDirs();
++
+     bool is_debug = handle_args( a );
+ 
+ #ifdef WITH_KF5DBus
+diff --git a/src/libcalamares/utils/Dirs.cpp b/src/libcalamares/utils/Dirs.cpp
+index f333d6e64..6118fb412 100644
+--- a/src/libcalamares/utils/Dirs.cpp
++++ b/src/libcalamares/utils/Dirs.cpp
+@@ -115,6 +116,14 @@ setXdgDirs()
+     s_haveExtraDirs = !( s_extraConfigDirs.isEmpty() && s_extraDataDirs.isEmpty() );
+ }
+ 
++void
++setNixosDirs()
++{
++    s_extraConfigDirs << "/run/current-system/sw/share/calamares/";
++    s_extraDataDirs << "/run/current-system/sw/share/calamares/";
```

Comment:
I would expect this to be covered by https://github.com/calamares/calamares/blob/baf8ef2c33879511bcc168cae03ab0c2aee1f505/src/libcalamares/utils/Dirs.cpp#L113

Reply:
That would make sense, I'll test it tomorrow and see, however since I made the patch in the first place I think at least at some point that was not enough

Reply:
Tested it without, seems like its necessary since by default it will look for only files within `calamares`'s nix store path, but since `calamares-nixos-extensions` is a seperate package, it can't find its `settings.conf` and will refuse to start

Reply:
Maybe it was just that it was missing from `s_extraConfigDirs`, since `XDG_CONFIG_DIRS` [looks to `/etc/xdg`](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html#variables), which will not have the files without them being added with `environment.etc` NixOS module. But unless I am misreading the code, I would expect `s_extraDataDirs` to be set correctly since `/run/current-system/sw/share` should already be in `XDG_DATA_DIRS`.

Reply:
I would expect it to work as well, `/run/current-system/sw/share` is in `XDG_DATA_DIRS`, and `XDG_DATA_DIRS` is searched in [Dirs.cpp](https://github.com/calamares/calamares/blob/69e00d181a00c375542f88e8d7fb32021575232f/src/libcalamares/utils/Dirs.cpp#L113), but for some reason it  isn't finding `/run/current-system/sw/share/calamares/settings.conf`. I think this is because `settings.conf` is searched with [`XDG_CONFIG_DIRS` instead](https://github.com/calamares/calamares/blob/69e00d181a00c375542f88e8d7fb32021575232f/src/libcalamares/utils/Dirs.cpp#L112).